delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,383 @@
+                                                self.systeminfo_command,
+                                                'storage')
+            enclosure_info = self.handle_detail_ssh(context,
+                                                    self.enclosure_command,
+                                                    'enclosure')
+            enclosure_res = enclosure_info.split('\n')
+            enclosure = enclosure_res[1].split(':')
+            serial_number = enclosure[7]
+            map = {}
+            self.handle_detail(systeminfo, map, split=' ')
+
+            status = map.get('statistics_status') == 'on' and 'normal' or \
+                'offline'
+            location = map.get('location')
+            free_capacity = self.handle_capacity(map.get(
+                'total_free_space'))
+            used_capacity = self.handle_capacity(map.get(
+                'total_used_capacity'))
+            raw_capacity = self.handle_capacity(map.get(
+                'total_drive_raw_capacity'))
+            subscribed_capacity = self.handle_capacity(map.get(
+                'total_allocated_extent_capacity'))
+            s = {
+                'name': map.get('name'),
+                'vendor': 'IBM',
+                'model': map.get('product_name'),
+                'status': status,
+                'serial_number': serial_number,
+                'firmware_version': map.get('code_level'),
+                'location': location,
+                'total_capacity': int(free_capacity + used_capacity),
+                'raw_capacity': int(raw_capacity),
+                'subscribed_capacity': int(subscribed_capacity),
+                'used_capacity': int(used_capacity),
+                'free_capacity': int(free_capacity)
+            }
+            return s
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage: %s" % (e.msg)
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def handle_detail(self, deltail_info, map, split):
+        detail_arr = deltail_info.split('\n')
+        for detail in detail_arr:
+            if detail is not None and detail != '':
+                strinfo = detail.split(split, 1)
+                key = strinfo[0]
+                value = ''
+                if len(strinfo) > 1:
+                    value = strinfo[1] and strinfo[1] or ''
+                map[key] = value
+
+    def list_storage_pools(self, context):
+        try:
+            pool_list = []
+            poolinfo = ''
+            poolinfo = self.handle_detail_ssh(context,
+                                              self.poolinfo_command, 'pool')
+            pool_res = poolinfo.split('\n')
+            i = 1
+            while i < len(pool_res):
```

Comment:
If len(pool_res) is 1, pool_list would be null?

Reply:
yes,the first line we do not use it